### PR TITLE
Sdk c4 68

### PR DESCRIPTION
**Pull request by sanchaymittal**

### CODE REVIEW

**Review by LayneHaber — APPROVED** [review]
Would like @rhlsthrm to review as well but lgtm!

File: packages/sdk/src/sdk.ts [hunk 1]
```diff
@@ -365,31 +377,80 @@ export class NxtpSdk {
     }
 
     const inbox = generateMessagingInbox();
-    const receivedResponsePromise = new Promise<AuctionResponse>((res, rej) => {
+
+    let AuctionBids: AuctionResponse[] = [];
```
Comment (rhlsthrm) [inline packages/sdk/src/sdk.ts:381]: Don't capitalize variable name like this.

File: packages/sdk/src/sdk.ts [hunk 2]
```diff
@@ -365,31 +377,80 @@ export class NxtpSdk {
...
+        } else {
+          AuctionBids.push(data);
+          if (AuctionBids.length >= 5) {
+            return;
+          }
         }
+      });
 
+      AuctionBids.sort((a, b) => {
+        return parseFloat(b.bid.amountReceived) - parseFloat(a.bid.amountReceived);
```
Comment (rhlsthrm) [inline packages/sdk/src/sdk.ts:402]: We can't parseFloat here, we are dealing with BigNumbers that could be larger than JS numbers.

File: packages/sdk/src/sdk.ts [hunk 3]
```diff
@@ -365,31 +377,80 @@ export class NxtpSdk {
...
+          this.logger.error(
+            { method, methodId, signer, receivingChainId, receivingAssetId, router: data.bid.router },
+            routerLiq.error.message,
+          );
+          return;
+        }
 
-        // TODO: compare multiple responses
+        // check if the price changes unfovorably by more than the slippage tolerance(percentage).
+        const priceImpact = ((parseFloat(data.bid.amountReceived) - parseFloat(amount)) / parseFloat(amount)) * 100;
```
Comment (rhlsthrm) [inline packages/sdk/src/sdk.ts:436]: Need to make this BigNumber math.

**Review by rhlsthrm — COMMENTED** [review]
Needs some fixes.

File: packages/sdk/src/sdk.ts (hunk 1 above)
Comment (sanchaymittal) [reply packages/sdk/src/sdk.ts:381]: done

**Comment by sanchaymittal** [comment]
fixes: https://github.com/connext/nxtp/pull/72

**Pull request merged**